kernel/mempool.c

```diff
@@ -9,11 +9,18 @@
 #include <zephyr/sys/math_extras.h>
 #include <zephyr/sys/util.h>
 
-static void *z_heap_aligned_alloc(struct k_heap *heap, size_t align, size_t size)
+typedef void * (sys_heap_allocator_t)(struct sys_heap *heap, size_t align, size_t bytes);
+
+static void *z_alloc_helper(struct k_heap *heap, size_t align, size_t size,
+			    sys_heap_allocator_t sys_heap_allocator)
 {
 	void *mem;
 	struct k_heap **heap_ref;
 	size_t __align;
+	k_spinlock_key_t key;
+
+	__ASSERT((align & (align - 1)) == 0,
+		"align must be a power of 2");
```

Comment:
I still think that a descriptive macro would be better here.
```suggestion
	__ASSERT(IS_POWER_OF_TWO(align));
```

Reply:
Can't use that as 0 must be a valid value too.